src/Microsoft.TestPlatform.Utilities/InferRunSettingsHelper.cs

```diff
@@ -103,6 +105,45 @@ public static void UpdateCollectSourceInformation(XPathNavigator runSettingsNavi
         }
 
         /// <summary>
+        /// Remove the <c>RunConfiguration.DesignMode</c> value for a run settings.
+        /// </summary>
+        public static void RemoveDesignMode(XPathNavigator runSettingsNavigator)
```

Comment:
We cant, generalized method (`RemoveNodeFromRunConfiguration`) is getting called from this method. We cant generalize because in that case we have to pass full path(`RunSettings/RunConfiguration/DesignMode`) and name of node to delete it and we want to have those info in this class only.